File: Linearsearch.py
def linear_search(arr, target):
    """
    Perform linear search on a list.
    Returns index if found, else -1.
    """
    for i, value in enumerate(arr):
        if value == target:
            return i
    return -1


if __name__ == "__main__":
    arr = [10, 20, 30, 40, 50]
    x = int(input("Enter number to search: "))
    print("Index:", linear_search(arr, x))

File: C/Linear_Search.c
#include <stdio.h>

int main() {
    int arr[100], n, target, i, found = 0;

    printf("Enter number of elements: ");
    scanf("%d", &n);

    printf("Enter %d integers:\n", n);
    for (i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    printf("Enter the number to search: ");
    scanf("%d", &target);

    for (i = 0; i < n; i++) {
        if (arr[i] == target) {
            found = 1;
            printf("✅ %d found at index %d.\n", target, i);
            break;
        }
    }

    if (!found) {
        printf("❌ %d not found in the array.\n", target);
    }

    return 0;
}